mp/src/game/client/momentum/ui/HUD/hud_measuretool.cpp

```diff
@@ -48,12 +20,7 @@ CON_COMMAND(showRuler, "Opens the ruler tool.\n")
         }
         else
         {
-            KeyValues* pKv = new KeyValues("Ruler Menu");
-            pKv->AddSubKey(new KeyValues("#MOM_Ruler_FirstPoint"));
-            pKv->AddSubKey(new KeyValues("#MOM_Ruler_SecondPoint"));
-            pKv->AddSubKey(new KeyValues("#MOM_Ruler_Measure"));
-            rulerMenu->ShowMenu(pKv, SelectMenuItem, OnClose);
-            pKv->deleteThis();
+            rulerMenu->ShowMenu("cfg/rulermenu.vdf", "Ruler Menu", OnClose);
```

Comment:
The problem I have with this is whenever users customize this file, but then re-install, or validate game files, or if we update the base file itself, this file gets overwritten with the defaults again. Let's instead:

1. Change these files we ship to be `<menuname>_default.vdf`
2. Change this method to only take the name of the menu (in this example, `"rulermenu"`) and the close method pointer.
   - Menu name should be stored in the menu under the key "menu_name"
   - Folder and extension will always be constant, and can be appended/constructed in the ShowMenu method
3. In ShowMenu, check if the non-default file exists and load it, otherwise copy the default file over to the non-default name, and then load it